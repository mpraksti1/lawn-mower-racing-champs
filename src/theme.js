const theme = {
  breakpoints: {
    mobile: 767,
    desktop: 768,
  },
  fonts: {
    baseFont: 'Helvetica',
    sansFont: 'Montserrat',
  },
  baseSizeUnit: 8,
  baseFontUnit: 16,
  colors: {
    yellow: '#f0ff0f',
    red: '#9d0404',
    gray: '#cfcfcf',
    darkGray: '#333',
    green: '#049d63',
    darkGreen: '#097149',
    white: '#eee',
    black: '#111',
    transparentBlack: 'rgba(0, 0, 0, 0.1)',
  },
  fontSizes: {
    xsm: '0.75rem',
    sm: '0.875rem',
    md: '1rem',
    lg: '1.125rem',
    xlg: '1.5rem',
    xxlg: '2.25rem',
  },
};

export default theme;
