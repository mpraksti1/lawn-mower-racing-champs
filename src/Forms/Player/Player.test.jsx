import { PlayerForm } from '.';
import theme from '../../theme';

describe('Player form component', () => {
  it('renders without crashing', () => {
    shallow(
      <PlayerForm
        history={fakeHistory}
        createPlayer={() => {}}
        alert={{}}
      />,
    );
  });

  it('matches snapshot', () => {
    const wrapper = shallowWithTheme(
      <PlayerForm
        history={fakeHistory}
        createPlayer={() => {}}
        alert={{}}
      />, theme,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
