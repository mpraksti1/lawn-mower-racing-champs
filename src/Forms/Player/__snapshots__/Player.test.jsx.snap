// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player form component matches snapshot 1`] = `
<BaseForm>
  <Formik
    component={[Function]}
    enableReinitialize={false}
    initialValues={
      Object {
        "first_name": "",
        "handedness": "right",
        "last_name": "",
        "rating": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": false,
        "_nodes": Array [
          "handedness",
          "rating",
          "last_name",
          "first_name",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "first_name": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "min": true,
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "handedness": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {
                "right",
                "left",
              },
              "refs": Map {},
            },
            "_whitelistError": [Function],
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "last_name": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "min": true,
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "rating": NumberSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "number",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  />
</BaseForm>
`;
